thunder/tests/test_networks.py

```diff
@@ -359,3 +359,54 @@ def test_quantization():
     assert len(sd) == len(sd2)
     for k, v in sd.items():
         assert_close(v, sd2[k])
+
+
+@thunder.tests.framework.requiresCUDA
+def test_thunderfx_mistral_nemo_small():
+    """
+    Runs a small version of Mistral-NeMo
+
+    This is largely based on code from Alexandros Koumparoulis.
+    """
+    import transformers
+
+    model_id = "mistralai/Mistral-Nemo-Base-2407"
+
+    # Setup a "small" version of NeMo-Mistral that does not require downloading
+    # weights. This is not a configuration that is worth benchmarking.
+    # This was created by using
+    #   MistralConfig(num_hidden_layers=1, max_position_embeddings=1024)
+    # and then manually diffing that returned object with:
+    #   transformers.AutoConfig.from_pretrained(model_id)
+    # until they lined up sans the hidden and embeddings changes, above.
+    config = transformers.models.mistral.configuration_mistral.MistralConfig(
+        num_hidden_layers=1,
+        torch_dtype=torch.bfloat16,
```

Comment:
This setting is ignored by the model instantiation. It can be checked by inspecting for example `model.model.layers[0].mlp.gate_proj.weight.dtype`.

@riccardofelluga, when you look into what Thunder executes for this and other HF models and what is missing for performance please update this test to use bfloat16 weights.